contracts/utils/ComplementaryServiceMetadata.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+/// @dev Service Registry interface
+interface IServiceRegistry {
+    enum ServiceState {
+        NonExistent,
+        PreRegistration,
+        ActiveRegistration,
+        FinishedRegistration,
+        Deployed,
+        TerminatedBonded
+    }
+
+    /// @dev Gets the service instance from the map of services.
+    /// @param serviceId Service Id.
+    /// @return securityDeposit Registration activation deposit.
+    /// @return multisig Service multisig address.
+    /// @return configHash IPFS hash pointing to the config metadata.
+    /// @return threshold Agent instance signers threshold.
+    /// @return maxNumAgentInstances Total number of agent instances.
+    /// @return numAgentInstances Actual number of agent instances.
+    /// @return state Service state.
+    function mapServices(uint256 serviceId) external view returns (uint96 securityDeposit, address multisig,
+        bytes32 configHash, uint32 threshold, uint32 maxNumAgentInstances, uint32 numAgentInstances, ServiceState state);
+
+    /// @dev Gets the owner of a specified service Id.
+    /// @param serviceId Service Id.
+    /// @return serviceOwner Service owner address.
+    function ownerOf(uint256 serviceId) external view returns (address serviceOwner);
+}
+
+/// @dev Provided zero address.
+error ZeroAddress();
+
+/// @dev Caught reentrancy violation.
+error ReentrancyGuard();
+
+/// @dev Account is unauthorized.
+/// @param account Account address.
+error UnauthorizedAccount(address account);
+
+/// @title ComplementaryServiceMetadata - Complementary Service Metadata contract to manage complementary service hashes
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract ComplementaryServiceMetadata {
```

Comment:
Please let's make this a proxy so we can easily update implementation if need be. Governance via DAO

Reply:
Or maybe we don't need that because we can just create new singletons

Reply:
Correct, the contract is owner-less, let's keep it this way, since we can specify the singleton with complementary metadata in the service metadata config.